workers/builder.py

```diff
@@ -101,11 +101,16 @@ def is_test_executable(src_dir: Path, name: str) -> bool:
 
     cargo('build', '-pneard', '--bin', 'neard', '--features',
           f'{test_feature},rosetta_rpc')
-    cargo('build',
-          '-pgenesis-populate',
-          '-prestaked',
-          '-pnear-test-contracts',
-          add_features=False)
+
+    build_additional_binaries_args = [
+        'build',
+        '-pgenesis-populate',
+        '-prestaked',
+        '-pnear-test-contracts'
+    ]
+    if spec.features and ("nightly" in spec.features.split(",")):
+        build_additional_binaries_args.append("--features=nightly")
+    cargo(*build_additional_binaries_args, add_features=False)
```

Comment:
Probably better to change `cargo` function to take `features` argument and have caller pass `spec.features` or `[feat for feat in spec.features if feat == 'nightly']`.

Reply:
good point, refactored that as suggested